modules/cas_cache/volume/vol_block_dev_top.c

```diff
@@ -868,8 +923,18 @@ int block_dev_create_exported_object(ocf_core_t core)
 
 	result = casdisk_functions.casdsk_exp_obj_create(dsk, dev_name,
 			THIS_MODULE, &_blockdev_exp_obj_ops);
-	if (!result)
-		bvol->expobj_valid = true;
+	if (result)
+		goto end;
+
+	bvol->expobj_valid = true;
+
+	bvol->expobj_wq = alloc_workqueue("expobj_wq%s-%s",
+			WQ_MEM_RECLAIM | WQ_HIGHPRI, 0,
+			get_cache_id_string(cache),
+			get_core_id_string(core));
+
+	if (!bvol->expobj_wq)
+		result = -ENOMEM;
```

Comment:
Shouldn't you add a rollback for `casdsk_exp_obj_create()` in case if `alloc_workqueue()` fails?

Reply:
That's right. Fixed.